tests/testsuite/build_script.rs

```diff
@@ -453,6 +453,41 @@ fn custom_build_env_var_rustc_linker() {
     p.cargo("build --target").arg(&target).run();
 }
 
+#[cargo_test]
+fn custom_build_env_var_rustc_linker_with_target_cfg() {
+    // Only run this test on aarch64, since it's difficult to construct
+    // a case suitable for all platforms.
+    if cross_compile::disabled() || cfg!(not(target_arch = "aarch64")) {
+        return;
+    }
```

Comment:
I lean towards testing on Linux for such reasons

* It's faster running the test suite on Linux than on macOS and Windows.
* Only on Linux Cargo's CI pipeline covers beta toolchain.
* There is no CI job running on `aarch64` at this moment. This is skipped.

```suggestion
// Only run this test on aarch64, since it's difficult to construct a case suitable for all platforms.
#[cargo_test]
#[cfg(target_os = "linux")]
fn custom_build_env_var_rustc_linker_with_target_cfg() {
    if cross_compile::disabled() {
        return;
    }
```

And below change to `i686` I believe it would work.

Reply:
Oh, good catch! Thanks!